src/test/java/training/chessington/model/pieces/KnightTest.java

```diff
@@ -1,7 +1,137 @@
 package training.chessington.model.pieces;
 
+import org.junit.Test;
+import training.chessington.model.Board;
+import training.chessington.model.Coordinates;
+import training.chessington.model.Move;
+import training.chessington.model.PlayerColour;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
 import static training.chessington.model.pieces.PieceAssert.*;
 import static org.assertj.core.api.Assertions.*;
 
 public class KnightTest {
+
+    @Test
+    public void knightCanMoveToCorrect8SpacesWhenInCentreOfBoard() {
+        Board board = Board.empty();
+        Piece knight = new Knight(PlayerColour.WHITE);
+        Coordinates position = new Coordinates(4,4);
+        board.placePiece(position, knight);
+
+        List<Move> moves = knight.getAllowedMoves(position, board);
+
+        List<Coordinates> correctDestinations = Arrays.asList(
+                new Coordinates(5, 2),
+                new Coordinates(6, 3),
+                new Coordinates(6, 5),
+                new Coordinates(5, 6),
+                new Coordinates(3, 2),
+                new Coordinates(2, 3),
+                new Coordinates(2, 5),
+                new Coordinates(3, 6)
+        );
+        List<Move> validMoves = correctDestinations.stream().map(destination -> new Move(position, destination)).collect(Collectors.toList());
+        assertThat(moves).containsAll(validMoves);
+    }
+
+    @Test
+    public void knightCannotMoveOutsideOfLShape() {
+        Board board = Board.empty();
+        Piece knight = new Knight(PlayerColour.WHITE);
+        Coordinates position = new Coordinates(4,4);
+        board.placePiece(position, knight);
+
+        List<Move> moves = knight.getAllowedMoves(position, board);
+
+        List<Coordinates> correctDestinations = Arrays.asList(
+                new Coordinates(5, 2),
+                new Coordinates(6, 3),
+                new Coordinates(6, 5),
+                new Coordinates(5, 6),
+                new Coordinates(3, 2),
+                new Coordinates(2, 3),
+                new Coordinates(2, 5),
+                new Coordinates(3, 6)
+        );
+        List<Move> validMoves = correctDestinations.stream().map(destination -> new Move(position, destination)).collect(Collectors.toList());
+        assertThat(validMoves).containsAll(moves);
+    }
+
+    @Test
+    public void knightCannotMoveMoreThan2SpacesInOneDirection() {
+        Board board = Board.empty();
+        Piece knight = new Knight(PlayerColour.WHITE);
+        Coordinates position = new Coordinates(7,4);
+        board.placePiece(position, knight);
+
+        List<Move> moves = knight.getAllowedMoves(position, board);
+
+        for (Move move: moves) {
+            assertThat(move.getTo().getRow()).isLessThan(8);

```

Comment:
What's the purpose of this test? I can't quite work out how its name matches its behaviour. In particular, all valid coordinates are less than 8, so the assertion is just checking that all the moves don't go off the edge of the board, right?

Reply:
Yes, that's me guilty of copying, pasting and forgetting to rewrite the details...